controllers/watchers/target_pod.go

```diff
@@ -0,0 +1,520 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package watchers
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/DataDog/chaos-controller/metrics"
+	"github.com/DataDog/chaos-controller/targetselector"
+	chaostypes "github.com/DataDog/chaos-controller/types"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	k8scache "sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DisruptionTargetHandler struct used to manage what to do when changes occur on the watched objects in the cache
+type DisruptionTargetHandler struct {
+	Recorder       record.EventRecorder
+	Reader         client.Reader
+	EnableObserver bool
+	Disruption     *v1beta1.Disruption
+	MetricsSink    metrics.Sink
+	Log            *zap.SugaredLogger
+}
+
+// On new target
+func (d DisruptionTargetHandler) OnAdd(obj interface{}) {
+	pod, okPod := obj.(*v1.Pod)
+	node, okNode := obj.(*v1.Node)
+
+	d.Log.Debugw("DisruptionTargetHandler ADD")
```

Comment:
Should we add name/namespace of the target and disruption name/namespace?

Reply:
Normally the disruption name /namespace is already injected [here](https://github.com/DataDog/chaos-controller/blob/main/controllers/disruption_controller.go#L101), but I think, it is a good idea to override the value here  to ensure the disruption name and namespace are valid in case the watcher is used by another package.